src/net_processing.cpp

```diff
@@ -427,6 +435,15 @@ void MaybeSetPeerAsAnnouncingHeaderAndIDs(NodeId nodeid, CConnman* connman) {
     }
 }
 
+bool TipMayBeStale(const Consensus::Params &consensusParams)
+{
+    AssertLockHeld(cs_main);
+    if (g_last_tip_update == 0) {
+        g_last_tip_update = GetTime();
+    }
+    return g_last_tip_update < GetTime() - consensusParams.nPowTargetSpacing * 3 && mapBlocksInFlight.empty();
```

Comment:
Since it is not uncommon for blocks to take more than 30 minutes to be found, maybe this threshold should be higher? It also seems like every node is going to be making that outbound connection at around the same time, maybe there should be some element of randomness to that to avoid some sort of fingerprinting?

Reply:
I think the scheduler behavior will mean that nodes are going to spread out their extra connections over a 10 minute window.  Since we already do feeler connections on a (poisson-random) 2 minute interval, and this is taking the place of a feeler, I don't think this will introduce too much network load.

If there's a fingerprinting concern, I can try to come up with some fuzzing on when the connection is made (maybe another poisson-random fuzz?), but I'm not sure I understand the specific fingerprinting risk with this logic?

Reply:
> but I'm not sure I understand the specific fingerprinting risk with this logic?

I don't think there is any. I had a fingerprinting concern but then realized that that concern was stupid and stopped being concerned about it.